From d292a44ea89faae9d1b095e0824a8b4db9833c7b Mon Sep 17 00:00:00 2001
From: "Slash.Huang" <slash.linux.c@gmail.com>
Date: Thu, 3 Apr 2025 13:21:22 +0800
Subject: [PATCH] [LOW_LEVEL_CONG] add panel check script  on boot

pin7(gpio3_1) : 0 -> hdmi -> true
pin7(gpio3_1) : 1 -> lvds -> false

pin99(gpio43_1) : 0 -> 60
pin99(gpio43_1) : 1 -> 10
---
 include/configs/rzg2l-apollo.h | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/include/configs/rzg2l-apollo.h b/include/configs/rzg2l-apollo.h
index c70e8c54..d2d7e1ca 100644
--- a/include/configs/rzg2l-apollo.h
+++ b/include/configs/rzg2l-apollo.h
@@ -60,13 +60,18 @@
 /* ENV setting */
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"set_lvds_type_65=setenv LVDS_TYPE 65 \0" \
+	"set_lvds_type_10=setenv LVDS_TYPE 10 \0" \
+	"chk_lvds_panel=if gpio input 99; then run set_lvds_type_65; else run set_lvds_type_10; fi \0" \
+	"set_lvds_panel=setenv panel_type apollo_lvds_${LVDS_TYPE}.dtb \0" \
+	"chk_panel_dtb=if gpio input 7; then setenv panel_type apollo_hdmi.dtb; else run chk_lvds_panel; run set_lvds_panel; fi \0" \
 	"bootm_size=0x10000000 \0" \
-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p1 \0" \
+	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p1 ${panel_type}\0" \
 	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p1 \0" \
 	"bootimage=booti 0x48080000 - 0x48000000 \0" \
 	"emmcload=ext4load mmc 0:1 0x48080000 boot/Image;ext4load mmc 0:1 0x48000000 boot/r9a07g044l2-apollo.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:1 0x48080000 boot/Image;ext4load mmc 1:1 0x48000000 boot/r9a07g044l2-apollo.dtb;run prodsdbootargs \0" \
-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
+	"sd1load=ext4load mmc 1:1 0x48080000 boot/Image;ext4load mmc 1:1 0x48000000 boot/${panel_type};run prodsdbootargs \0" \
+	"bootcmd_check=run chk_panel_dtb; if mmc dev 1; then run sd1load; else run emmcload; fi \0"
 
 #define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
 
-- 
2.25.1

